tests/test_extractors.py

```diff
@@ -91,3 +91,15 @@ def test_extract_from_directory(self, extractor, tmp_path, file):
     def test_extract_from_nonsense(self, extractor):
         with pytest.raises(NotImplementedError):
             extractor.extract(8)
+
+    @pytest.mark.parametrize(
+        "source",
+        [
+            """
+                def foo():
+            """,
+        ],
+    )
+    def test_extract_with_syntax_error(self, extractor, file_path, file, caplog):
```

Comment:
**suggestion (testing):** Test case for handling multiple files with syntax errors is missing.

The PR introduces functionality to skip files with syntax errors during directory processing. It would be beneficial to include a test case that processes a directory containing multiple files, some of which have syntax errors, to ensure that files with valid syntax are still processed correctly.